src/sql-parser/tests/testdata/scalar

```diff
@@ -83,11 +83,7 @@ Op { op: "!~*", expr1: Identifier([Ident("a")]), expr2: Some(Value(String("foo")
 parse-scalar
 a !x 'foo'
 ----
-error:
-Parse error:
-a !x 'foo'
-  ^
-Unrecognized token
+Op { op: "!", expr1: Identifier([Ident("a")]), expr2: Some(Identifier([Ident("x")])) }
```

Comment:
Does it matter that the `foo` isn't represented in the output? Should `parse-scalar` do some kind of round tripping (like parse the AST's .to_string)? Probably a good idea to make sure tokens aren't getting dropped on the ground.

Reply:
Eep, good catch! I changed `parse_expr` to return an error if there are any tokens remaining after the parse, which I believe is the correct behavior. (Nothing calls that function besides tests.)